enclave/server.ts

```diff
@@ -51,11 +46,64 @@ const nStates = new ethers.Contract(
     signer
 );
 
+type ClaimedMove = {
+    uFrom: Tile;
+    uTo: Tile;
+};
+
 /*
  * Enclave's internal belief on game state stored in Board object.
  */
 let b: Board;
 
+/*
+ * Bijection between player's public keys and their socket IDs.
+ */
+let idToPubKey = new Map<string, string>();
+let pubKeyToId = new Map<string, string>();
+
+/*
+ * Moves claimed by players. A move is finalized whenever NewMove event is
+ * emitted.
+ */
+let claimedMoves = new Map<string, ClaimedMove>();
+
+/*
+ * Dev function for spawning a player on the map.
+ */
+async function spawn(
+    socket: Socket,
+    l: Location,
+    reqPlayer: Player,
+    sigStr: string
+) {
+    // If player is already spawned in, their socket ID will already be stored
+    if (idToPubKey.has(socket.id)) {
+        return;
+    }
+
+    const h = Player.hForSpawn(Utils.asciiIntoBigNumber(socket.id));
+    const sig = Utils.unserializeSig(sigStr);
+    if (sig && reqPlayer.verifySig(h, sig)) {
+        const pubkey = reqPlayer.bjjPub.serialize();
+
+        // Spawn if player is connected for the first time
+        if (!pubKeyToId.has(pubkey)) {
```

Comment:
could we avoid putting this much weight on the socketID 

i think about this bijection as a convenience structure so we know who we're communicating with, but it shouldn't control our belief of whether someone's already spawned into the game

Reply:
the bijection is a temporary structure that shifts all the time between player sessions 

information that is attached to the player, not the session, shouldn't be associated with the socketID. being spawned in is an example of something that should be attached directly to the player